.github/workflows/goth.yml

```diff
@@ -0,0 +1,80 @@
+name: Goth (PR and push)
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      # - <your-branch>    # put your branch name here to test it @ GH Actions
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  goth-tests:
+    name: Run integration tests
+    runs-on: goth
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Configure python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8.0'
+
+      - name: Configure poetry
+        uses: Gr1N/setup-poetry@v8
+        with:
+          poetry-version: 1.2.2
+
+      - name: Install dependencies
+        run: |
+          poetry env use python3.8
+          poetry install
+
+      - name: Disconnect Docker containers from default network
+        continue-on-error: true
+        # related to this issue: https://github.com/moby/moby/issues/23302
+        run: |
+          docker network inspect docker_default
```

Comment:
maybe it's time we included this into goth proper as a helper script? so that it doesn't have to be repeated in each repository that uses goth? adding an issue...